config/application.rb

```diff
@@ -31,5 +31,11 @@ class Application < Rails::Application
     # Application configuration should go into files in config/initializers
     # -- all .rb files in that directory are automatically loaded.
     config.autoload_paths += %W(#{config.root}/lib)
+
+    if ENV['IFRAME_ALLOW_ORIGIN'].present?
+      config.action_dispatch.default_headers = {
```

Comment:
Not sure if it matters, but won't this disable the other defaults (i.e `X-Content-Type-Options` and `X-XSS-Protection`) - `merge` instead..?